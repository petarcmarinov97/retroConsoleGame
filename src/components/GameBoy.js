import React, { Component } from 'react';
import Display from './Display';
import { Typography } from 'antd';
import Controller from './Controller';

export default class GameBoy extends Component {

    render() {
        const { Title } = Typography;
        const { gameTitle } = this.props;

        return <div className="game_boy">
            <>
                <Title level={4}>{gameTitle}</Title>
                <Display />
                <Controller />
            </>
        </div>;
    }
}
