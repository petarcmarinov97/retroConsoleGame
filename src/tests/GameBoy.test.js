import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from "enzyme-to-json";
import GameBoy from '../components/GameBoy';

describe('GameBoy tests', () => {

    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <GameBoy gameTitle="Tetris" />);
    })

    describe('Snapshots', () => {
        it("should render correctly", () => {
            expect(shallowToJson(wrapper)).toMatchSnapshot();
        })
    })
})