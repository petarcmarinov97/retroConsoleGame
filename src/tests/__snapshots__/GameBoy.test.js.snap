// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoy tests Snapshots should render correctly 1`] = `
<div
  className="game_boy"
>
  <Title
    gameTitle="Tetris"
  />
  <Display />
  <Controller />
</div>
`;
